src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import * as telemetry from '../../shared/telemetry/telemetry'
+import {
+    CloudWatchLogsData,
+    CloudWatchLogsGroupInfo,
+    CloudWatchLogsParameters,
+    LogStreamRegistry,
+    filterLogEventsFromUriComponents,
+} from '../registry/logStreamRegistry'
+import { createQuickPick, DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+import { Wizard } from '../../shared/wizards/wizard'
+import { createInputBox } from '../../shared/ui/inputPrompter'
+import { createURIFromArgs } from '../cloudWatchLogsUtils'
+import { DefaultCloudWatchLogsClient } from '../../shared/clients/cloudWatchLogsClient'
+import { CloudWatchLogs } from 'aws-sdk'
+
+export async function searchLogGroup(registry: LogStreamRegistry): Promise<void> {
+    let result: telemetry.Result = 'Succeeded'
+    const regionCode = 'us-west-2'
+    const client = new DefaultCloudWatchLogsClient(regionCode)
+    const logGroups = await logGroupsToArray(client.describeLogGroups())
+    const response = await new SearchLogGroupWizard(logGroups).run()
+    if (response) {
+        const logGroupInfo: CloudWatchLogsGroupInfo = {
+            groupName: response.logGroup,
+            regionName: regionCode,
+        }
+
+        const parameters: CloudWatchLogsParameters = {
+            limit: registry.configuration.get('limit', 10000),
+            filterPattern: response.filterPattern,
+        }
+
+        const uri = createURIFromArgs(logGroupInfo, parameters)
+        const initialStreamData: CloudWatchLogsData = {
+            data: [],
+            parameters: parameters,
+            busy: false,
+            logGroupInfo: logGroupInfo,
+            retrieveLogsFunction: filterLogEventsFromUriComponents,
+        }
+
+        await registry.registerLog(uri, initialStreamData)
+        const doc = await vscode.workspace.openTextDocument(uri) // calls back into the provider
+        vscode.languages.setTextDocumentLanguage(doc, 'log')
+        await vscode.window.showTextDocument(doc, { preview: false })
+    } else {
+        result = 'Cancelled'
+    }
+    telemetry.recordCloudwatchlogsOpenStream({ result })
```

Comment:
Next PR let's try to merge this logic with the other command somehow (perhaps combine the wizard?)
`viewLogStream` and `searchLogGroup` only really differ by parameters which is what the wizard provides

Reply:
I'll add an issue for this. Not sure about the best way to do it at the moment, but I can start thinking about it.